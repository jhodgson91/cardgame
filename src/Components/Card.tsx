import React, { FC } from 'react'
import './Main.scss'

//Define props types
export interface Props {
    image: string;
    value: string;
    suit: string;
    code: string;
}

const Card: FC<Props> = ({ image, value, suit, code }) => {
    return (
        <div className="card small-4">
            <img className="card-img" src={image}/>
        </div>
    )
}

export default Card
