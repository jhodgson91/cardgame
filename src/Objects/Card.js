
class Card
{
    constructor(data)
    {
        this.data = data
    }

    get image() {
        return this.data.image;
    }
    get value() {
        return this.data.value;
    }
    get suit() {
        return this.data.suit;
    }
    get code() {
        return this.data.code;
    }
    
}

export default Card