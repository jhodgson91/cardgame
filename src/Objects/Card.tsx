export interface CardData {
    image: string,
    value: string,
    suit: string,
    code: string
}

class Card
{
    data: CardData

    constructor(data:CardData)
    {
        this.data = data
    }

    get image() {
        return this.data.image;
    }
    get value() {
        return this.data.value;
    }
    get suit() {
        return this.data.suit;
    }
    get code() {
        return this.data.code;
    }
}

export default Card