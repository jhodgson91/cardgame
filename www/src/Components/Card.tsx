import * as React from 'react';
import './Main.scss'

//Define props types
export interface Props {
    image: string;
    value: string;
    suit: string;
    code: string;
}

export default class Card extends React.Component<Props> {

    constructor(props: Props) {
        super(props);
    }

    render(): React.ReactNode {
        return (
            <div className="card small-1">
                <img className="card-img" src={this.props.image} />
            </div>
        );
    }
}
