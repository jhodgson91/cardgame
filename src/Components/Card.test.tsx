import * as React from 'react';
import * as enzyme from 'enzyme';
import Card from './Card';

it('renders with an image and a key', () => {
  const card = enzyme.shallow(<Card image="https://deckofcardsapi.com/static/img/AH.png"/>);
  expect(card.find('img').prop('src')).toBe('https://deckofcardsapi.com/static/img/AH.png');
});

//Not sure we want this to work?
it('renders without an image', () => {
  const card = enzyme.shallow(<Card />);
  expect(card.find('img').prop('image')).toBeUndefined;
});
