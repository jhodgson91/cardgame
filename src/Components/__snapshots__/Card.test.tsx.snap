// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the same as last time 1`] = `
<div
  className="card small-4"
>
  <img
    className="card-img"
    src="https://deckofcardsapi.com/static/img/KH.png"
  />
</div>
`;
